sway-core/src/semantic_analysis/ast_node/expression/intrinsic_function.rs

```diff
@@ -541,6 +541,60 @@ impl TypedIntrinsicFunctionKind {
                 let return_type = insert_type(TypeInfo::Tuple(vec![]));
                 (intrinsic_function, return_type)
             }
+            Intrinsic::Add | Intrinsic::Sub | Intrinsic::Mul | Intrinsic::Div => {
```

Comment:
I noticed this in other places in this file as well but it is possible to pass in type arguments to these intrinsics and the compiler won't complain. Is that correct? Maybe we should add a check here that the number of type arguments is exactly zero.

Reply:
> I noticed this in other places in this file as well but it is possible to pass in type arguments to these intrinsics and the compiler won't complain. Is that correct? Maybe we should add a check here that the number of type arguments is exactly zero.

I've added this now.

Reply:
> Long term, I think we should let the type checker proper handle this by exposing the signature of this intrinsic in the `core` library itself, e.g.,
> 
> ```rust
> #[intrinsic]
> fn my_special_function(...);
> ```
> 
> Note the `;` here -- the intrinsic is assigned no body. The type checker will see that this is `#[intrinsic]` and therefore allow the body-less function. However, it will collect the signature of the function and any calls to it will consult that signature. The only special code in the compiler is then recognizing to the standard library item and assigning it dynamic semantics (fairly deep in the compiler, in IR generation, with type substitutions etc. already known and with the sound assumption that there are no inference variables left).

I like this suggestion. Can you also post this in the RFC for intrinsics? I (and @otrho) had suggested a table based approach but this could be better.